src/odemis/util/angleres.py

```diff
@@ -87,76 +84,114 @@ def _extractAngleInformation(data, hole, dtype=None):
-        # Each px in the raw data corresponds to a specific theta-phi combination
-        theta, phi, omega = _FindAngle(data, xpix, ypix, pixel_size)
-
-        # theta_data: array containing theta values for each px in raw data
-        theta_data[i, :] = theta
-        # phi_data: array containing phi values for each px in raw data
-        phi_data[i, :] = phi
-
-        # intensity_data contains the intensity values from raw data.
-        # It already reflects the shape of the mirror
-        # and is normalized by omega (solid angle:
-        # measure for photon collection efficiency depending on theta and phi)
-        intensity_data[i, :] = cropped_image[i] / omega
+    x_indices = numpy.linspace(0, data.shape[1] - 1, data.shape[1])  # list of px indices
+    # correct x coordinates to be 0 at the hole position
+    x_pos = x_indices - pole_pos[0]  # x coordinates of the pixels (horizontal)
+
+    y_indices = numpy.linspace(0, data.shape[0] - 1, data.shape[0])  # list of px indices
+    # correct y coordinates to be 0 at the hole position
+    y_pos = (y_indices - pole_pos[1]) + (2 * parabola_f) / pixel_size[0]  # y coordinates of the pixels (vertical)
+
+    # create two arrays with set of x and y coordinates and with same shape of data
+    x_array, y_array = numpy.meshgrid(x_pos, y_pos)
+
+    # TODO can we modify xpos and ypos so we only search for angles within half circle
+    # Finds the angles of emission by the sample for each px (x,y) in the raw data.
+    # These angles only depend on the mirror geometry.
+    # Each px in the raw data corresponds to a specific theta-phi combination
+    # theta_data: array containing theta values for each px in raw data
+    # phi_data: array containing phi values for each px in raw data
+    theta_data, phi_data, omega = _FindAngle(x_array, y_array, pixel_size, parabola_f)
+
+    # intensity_data contains the intensity values from raw data.
+    # It already reflects the shape of the mirror
+    # and is normalized by omega (solid angle:
+    # measure for photon collection efficiency depending on theta and phi)
+    intensity_data = cropped_image / omega
 
     return theta_data, phi_data, intensity_data, circle_mask_dilated
 
 
-def AngleResolved2Polar(data, output_size, hole=True, dtype=None):
+def _FindAngle(x_array, y_array, pixel_size, parabola_f):
+    """
+    For given pixels, finds the angle of the corresponding ray
+    :parameter x_array: (ndarray) x coordinates of the pixels
+    :parameter y_array: (ndarray) y coordinate of the pixel
+    :parameter pixel_size: (float, float) detector pixel size (X/Y)
+    :parameter parabola_f: (float) parabola_parameter=1/(4f): f: focal point of mirror (place of sample)
+    :returns: (ndarray) theta, phi (the corresponding spherical coordinates for each pixel in detector)
+                              and omega (solid angle: angular range collected per px)
+    """
+
+    y = x_array * pixel_size[0]
```

Comment:
What's the idea behind naming the X data `y`, and the Y data `z`? Maybe if you were using entirely different letters, it'd be less confusing. Or add a comment explaining.

Reply:
yes that also gives me still headaches. I know that it had something to do where the axes regarding the mirror are defined, but I am not sure anymore which one was what. I renamed them now and put a note there. I will ask Toon when I am back and will update it then.